matchms/filtering/correct_charge.py

```diff
@@ -14,17 +15,21 @@ def correct_charge(spectrum_in: SpectrumType) -> SpectrumType:
     spectrum = spectrum_in.clone()
 
     ionmode = spectrum.get("ionmode", None)
+    if ionmode:
+        assert ionmode == ionmode.lower(), ("Ionmode field not harmonized.",
+                                            "Apply 'make_ionmode_lowercase' filter first.")
 
     charge = spectrum.get("charge", None)
+    assert not isinstance(charge, str), ("Charge is given as string.",
```

Comment:
would be nicer to use numpy.isscalar here

https://numpy.org/doc/stable/reference/generated/numpy.isscalar.html

Reply:
Not sure. I wanted to specifically return a message if a string is given, which happens when importing spectra from GNPS json files without running the ``make_charge_scalar`` filter first.

Reply:
To me this is easier to read and understand.
`assert numpy.isscalar(charge)`

Reply:
``numpy.iscalar`` won't work. It only checks if it is a scalar and not an array. 
``numpy.iscalar("text")`` will also return ``True``

Reply:
@fdiblen are you fine with leaving it as it is and merging it?

Reply:
Numpy isscalar indeed does not seem to do what its name implies. It is not suited here, as Florian explained and can be read here (https://numpy.org/devdocs/reference/generated/numpy.isscalar.html).
The documentation states:
'If you need a stricter way to identify a numerical scalar, use isinstance(x, numbers.Number), '
So that would be another option. 
In this case though, you want to help users who make a common mistake of inputting a string specifically. I think it makes sense to keep it like this.

Reply:
Thanks @cwmeijer, to me that resolves it. I would go ahead and merge this soon then.